openmc/particle_restart.py

```diff
@@ -43,28 +39,49 @@ def __init__(self, filename):
         import h5py
         self._f = h5py.File(filename, 'r')
 
-        # Read all metadata
-        self._read_data()
+        # Ensure filetype and revision are correct
+        if 'filetype' not in self._f or self._f[
+                'filetype'].value.decode() != 'particle restart':
+            raise IOError('{} is not a particle restart file.'.format(filename))
+        if self._f['revision'].value != 1:
+            raise IOError('Particle restart file revision is not consistent.')
```

Comment:
I think this error message (actually, a lot of our errors messages) would be very opaque to the novice user. Perhaps include the file's revision along with revision for the currently installed version of OpenMC to clarify this message.